src/main/java/io/cdap/plugin/googleads/source/batch/GoogleAdsBatchSourceConfig.java

```
@@ -155,36 +158,70 @@ protected void validateAuthorisation(FailureCollector failureCollector, GoogleAd
-    if (containsMacro(REPORT_TYPE)){
+    if (containsMacro(REPORT_TYPE)) {
       return;
     }
     ReportDefinitionReportType reportDefinitionReportType = null;
     try {
       reportDefinitionReportType = getReportType();
     } catch (IllegalArgumentException ex) {
-      failureCollector.addFailure("Invalid reportDefinitionReportType", "Enter valid reportDefinitionReportType").withConfigProperty(REPORT_TYPE);
+      failureCollector.addFailure("Invalid reportDefinitionReportType",
+                                  "Enter valid reportDefinitionReportType")
+        .withConfigProperty(REPORT_TYPE);
     }
-    if (reportDefinitionReportType != null){
+    if (reportDefinitionReportType != null) {
       validateFields(failureCollector, googleAdsHelper);
     }
   }
 
   protected void validateFields(FailureCollector failureCollector, GoogleAdsHelper googleAdsHelper) {
-    if (containsMacro(REPORT_FIELDS)){
+    if (containsMacro(REPORT_FIELDS)) {
       return;
     }
+    List<String> reportFields = getReportFields();
+    Set<String> reportFieldsSet = new HashSet<>(reportFields);
+    if (reportFieldsSet.size() != reportFields.size()) {
+      failureCollector.addFailure("reportFields contains duplicates",
+                                  "Enter valid reportFields according to record type")
+        .withConfigProperty(REPORT_FIELDS);
+    }
+
+    ReportDefinitionField[] reportDefinitionFields;
     try {
-      List<String> fields = googleAdsHelper.getAllFields(this);
-      if (!fields.containsAll(getReportFields())){
-        failureCollector.addFailure("Invalid reportFields", "Enter valid reportFields according to record type").withConfigProperty(REPORT_FIELDS);
-      }
+      reportDefinitionFields = googleAdsHelper.getReportDefinitionFields(this);
     } catch (OAuthException | ValidationException | RemoteException ignored) {
+      return;
```

Never ignore exceptions. This should be a validation error.